Chess/include/Queen.h

```diff
@@ -0,0 +1,14 @@
+#include "Bishop.h"
+#include "Rook.h"
+
+
+class Queen : public Rook, public Bishop
```

Comment:
Review the inheritance hierarchy:the class Queen is inheriting from both Rook and Bishop. Since Rook and Bishop are virtual base classes of Piece, you may need to modify the inheritance hierarchy to avoid the "diamond problem" of multiple inheritance. Ensure that the inheritance hierarchy is designed correctly based on the desired behavior.
In addition, why does the queen inherit from rock and bishop? 
Remember the following rule:
We use inheritance if we can say the following sentence: dog 'is a' type of animal.
Can you say that about the queen? the queen is a rock?...